### ISSUE TITLE
The experimental web version is not working

### ISSUE THREAD

**Issue by hakro** [YuriSizov/gdsion#15]
The web version that uses the latest Godot 4.3 web features is not working

https://humnom.net/demos/gdsion/

It gets stuck in the Splash screen, then after some time, a message appears, saying that the tab is slowing down FF.

![Screenshot_20240517_110050](https://github.com/YuriSizov/gdsion/assets/3624853/3cf19db4-56e9-4a4a-a5c6-d40ae7edbaea)

My specs : 
Linux 6.5.0
KDE 6
Firefox 126.0

**Comment by YuriSizov** [comment]
@hakro It doesn't work in Firefox and Safari due to engine issues. There is nothing I can do about it. Only Chrome-likes are supported by Godot web exports.

**Comment by hakro** [comment]
Thank you for the clarification @YuriSizov 
Indeed, I tested with Brave, the app loaded, and I could play the piano for a second, but then it froze.
But after reloading the page, it all worked pretty well.

**Comment by YuriSizov** [comment]
> I could play the piano for a second, but then it froze.

Hmm, that shouldn't happen. Did you happen to catch a glimpse of the developer console? If this happens again, please do. There are going to be a bunch of unrelated warnings, but perhaps also something valuable about the source of this freeze.

**Comment by hakro** [comment]
Yeah, Tried a couple of times, there seems to be a 50-50 chance that the Brave tab freezes. In both cases, the console logs are similar. When it freezes, there are no additional logs after the line that shows the Godot and GLES versions.
When it freezes, it's really just a second after the keyboard is available, I even had time to play a note or too.

In the screenshot, it's frozen on that A note, and the spectrum analyzer is also frozen in that state. Can't do anything but close the whole browser.

![Screenshot_20240517_162420](https://github.com/YuriSizov/gdsion/assets/3624853/f2db1a0e-68db-4272-8d19-3b200bf70495)

Brave version : Version 1.66.108 Chromium: 125.0.6422.41 (Official Build)

**Comment by YuriSizov** [comment]
Thanks, I'll try to find some time to look more into this!

**Comment by djtuBIG-MaliceX** [comment]
Just for the record, have tried WASM build with Firefox 136.0.1 (64-bit) on Ubuntu 22.04.5 LTS x64, an Windows 11 x64 (Build 26100) and appears to be working fine. :)  (Also on Android Chrome/Samsung Internet and Firefox)

For Safari, at least on safari mobile (iPhone 11 iOS 17.6.1) it just hangs on the godot loading page.

**Comment by YuriSizov** [comment]
Yeah, it should work in Firefox and any Chromium-based browser these days. (Although I don't keep the hosted version of the example app up to date, so it may still exhibit some old issues.)

Safari isn't going to work because Apple heavily limits the amount of memory that a wasm app can use. It's around 1-1.5 gigabytes (it's not fixed, but is somewhere in that range). And Godot's web exports easily reach around 2 gigabytes. In Bosca Ceoil Blue [I've implemented a custom web shell](https://humnom.net/thoughts/676e811c-bosca-for-web-better-landing-page.html) to handle these issues more gracefully (it still fails in the end, but you might just be able to see the UI). I can do something similar to the example app, but ultimately, web exports will be severely limited until something changes (either Apple relaxes its constraints, or Godot becomes more lean on web).

A custom Godot export could also help, stripping unused modules. Though that's out of scope of this extension, of course. I should document these caveats somewhere in this repo.
